cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/template/etl/realtime/RealtimeElasticsearchSinkTest.java

```diff
@@ -138,7 +138,7 @@ public void testESSink() throws Exception {
       AdapterManager adapterManager = createAdapter(adapterId, adapterConfig);
 
       adapterManager.start();
-      TimeUnit.SECONDS.sleep(5);
+      TimeUnit.SECONDS.sleep(10);
```

Comment:
instead of sleeping for random interval, can we check for the success condition sleeping every 1 seconds upto a specified retry. If the success criteria is not met after the n retries then fail the test. Sleeping for 10 seconds may fail if the build box is slower. For example see HelloWorldTest

```
 RuntimeMetrics metrics = flowManager.getFlowletMetrics("saver");

 metrics.waitForProcessed(5, 5, TimeUnit.SECONDS);
```